bootloader/assets/grub.go

```diff
@@ -144,6 +151,15 @@ menuentry 'System setup' --hotkey=f 'uefi-firmware' {
 `
 
 func init() {
-	registerInternal("grub.cfg", []byte(grubBootConfig))
-	registerInternal("grub-recovery.cfg", []byte(grubRecoveryConfig))
+	// static command line for edition 1 of grub config
+	registerInternal("grub.cfg:edition=1:static_cmdline", []byte(grubBootConfigStaticCmdline))
```

Comment:
this is kind of cute but we would expect the same command line to be reused across possibly many editions, so this might get unwieldy quickly.  I suggest to try something a bit more complicated like this:

```
registerSnippetForEditions("grub.cfg:static-cmdline", []forEditions{
   {FirstEdition: 1, Snippet: []byte(..A..)},
   {FirstEdition: 5, Snippet: []byte(..B..)},
})
```

this would register a first variant used by edition 1-4,  and second variant to be used starting from edition 5.

There would be a getter that takes the key and edition.

The tests should have a check that the Snippet for the current edition of an asset matches what is inside the asset.

Reply:
@bboozzoo as I mentioned on IRC, it's fine to keep the old storage approach for assets, and have a different one for snippets